### PR TITLE
[AUD-1728] Use drawer nav to support notifs

### PR DESCRIPTION
**Pull request by raymondjacobson**
### Description

Switch notifications drawer to react navigation DrawerNavigation. This supports some better animation and lets us do more complex navigation things.

Note: This PR drops chromecast support (which was broken anyway). Had to upgrade the version for compat with crashlytics, etc., which I upgraded in this PR. It's kind of a no-op, but a subsequent PR will fix it using the new APIs.

https://user-images.githubusercontent.com/2731362/158954914-a101f043-6fdf-49a6-b795-d82035bd3f3a.mp4

### Dragons

*Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?*

### How Has This Been Tested?

*Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.*

Simulator vs. staging

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

Staging test flight

### CODE REVIEW

File: packages/mobile/src/hooks/useNavigation.ts [hunk 1]
```diff
@@ -48,7 +58,10 @@ export const useNavigation = <
   return useMemo(
     () => ({
       navigate: performNavigation(nativeNavigation.navigate),
-      push: performNavigation(nativeNavigation.push),
+      push:
+        'push' in nativeNavigation
+          ? performNavigation(nativeNavigation.push)
+          : () => {},
```
Comment (sliptype) [inline packages/mobile/src/hooks/useNavigation.ts:64]: Maybe we should log something here? Might be confusing to someone calling `push` and nothing happening

File: packages/mobile/src/screens/root-screen/RootScreen.tsx [hunk 2]
```diff
@@ -34,15 +84,23 @@ export const RootScreen = () => {
     (signedIn && !onSignUp) ||
     isAccountAvailable
 
-  return (
-    <Stack.Navigator
-      screenOptions={{ gestureEnabled: false, headerShown: false }}
+  return isAuthed ? (
```
Comment (sliptype) [inline packages/mobile/src/screens/root-screen/RootScreen.tsx:87]: So clean

**Review by sliptype — APPROVED** [review]
Amazing! Super excited for this

File: packages/mobile/src/screens/root-screen/RootScreen.tsx (hunk 2 above)
Comment (raymondjacobson) [reply packages/mobile/src/screens/root-screen/RootScreen.tsx:87]: yay!

File: packages/mobile/src/hooks/useNavigation.ts (hunk 1 above)
Comment (raymondjacobson) [reply packages/mobile/src/hooks/useNavigation.ts:64]: great idea

**Pull request merged**